src/GitTools.jl

```diff
@@ -65,22 +65,43 @@ function transfer_progress(progress::Ptr{LibGit2.TransferProgress}, p::Any)
     return Cint(0)
 end
 
+const GIT_REGEX =
+    r"^(?:(?:git|ssh|https)(?:://|@))?(?:[\w\.\+\-:]+@)?([^/:]+)[:/](.+?)(?:\.git)?$"i
```

Comment:
I think it makes more sense to use named groups at this point. Also it is weird that `git@` is captured by the scheme part of the regex. I would recommend this:
```julia
r"^(?:(?<scheme>git|ssh|https)://)?(?:[\w\.\+\-:]+@)?(?<hostname>[^/:]+)(?(<scheme>)/|:)(?<path>.+?)(?:\.git)?$"
```